flecsi/exec/local/bind_prologue.hh

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2016, Triad National Security, LLC
+// All rights reserved.
+
+#ifndef FLECSI_EXEC_LOCAL_BIND_PROLOGUE_HH
+#define FLECSI_EXEC_LOCAL_BIND_PROLOGUE_HH
+
+#include <flecsi-config.h>
+
+#include "flecsi/data/privilege.hh"
+#include "flecsi/exec/local/policy.hh"
+#include "flecsi/flog.hh"
+#include "flecsi/util/annotation.hh"
+#include "flecsi/util/demangle.hh"
+
+#include <functional>
+#include <tuple>
+
+namespace flecsi {
+inline flog::devel_tag bind_prologue_tag("bind_prologue");
+}
+
+// task_prologue is implemented per backend:
+#if FLECSI_BACKEND == FLECSI_BACKEND_legion
+#include "flecsi/exec/leg/bind_accessors.hh"
+#elif FLECSI_BACKEND == FLECSI_BACKEND_mpi
+#error "The MPI backend has no need for bind_accessors"
+#elif FLECSI_BACKEND == FLECSI_BACKEND_hpx
+#include "flecsi/exec/hpx/bind_accessors.hh"
+#endif
+
+namespace flecsi {
+namespace exec {
+
+#ifdef DOXYGEN // implemented per-backend
+/// Handling for low-level special task parameters/arguments.
+/// The exact member function signatures may vary between backends.
+template<task_processor_type_t Proc>
+struct bind_accessor {
+protected:
+  /// Constructible from backend-specific arguments.
+  template <typename...Args>
+  bind_accessor(Args&&... args);
+
+  /// Send a raw field reference to a raw accessor.
+  template<typename T, Privileges P>
+  void visit(data::accessor<data::raw, T, P> &);
+};
+#endif
+
+template<task_processor_type_t ProcessorType>
+struct bind_prologue : bind_accessors<ProcessorType> {
```

Comment:
I'm also confused by the directory name: `exec/local/tracer.hh` sort of makes sense, since it happens that the set of cross-node backends is exactly the same as the set of backends that do tracing.  However, there's nothing "local" about this shared `bind_parameters` code; in fact, it's more or less "distant" in that the parameters are being (further) initialized away from the arguments.  Once upon a time we had `flecsi/execution/common`, but maybe we should just put these files loose in `exec` (like `prolog.hh`) even though not _all_ backends use them?